cunumeric/array.py

```diff
@@ -1139,12 +1139,89 @@ def cumsum(self, axis=None, dtype=None, out=None):
         )
         return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
 
-    @unimplemented
-    def diagonal(self, offset=0, axis1=0, axis2=1):
-        numpy_array = self.__array__(stacklevel=3).diagonal(
-            offset=offset, axis1=axis1, axis2=axis2
+    # diagonal helper. Will return diagonal for arbitrary number of axes;
+    # currently offset option is implemented only for the case of number of
+    # axes=2. This restriction can be lifted in the future if there is a
+    # use case of having arbitrary number of offsets
+    def diag_helper(self, offset=0, axes=None, extract=True):
+        N = len(axes)
+        # diag_helper can be used only for arrays with dim>=1
+        if self.ndim < 1:
+            raise RuntimeError("diag_helper is implemented for dim>=1")
+        elif self.ndim == 1:
+            m = self.shape[0] + np.abs(offset)
```

Comment:
Throw a ValueError here if `axes is not None` (doesn't make sense to specify axes if `extract == False`).